### DISCUSSION TITLE
recommendation to use pagination from prisma1

### DISCUSSION THREAD

**Discussion by alan345** [prisma/prisma#3344] [Q&A] [answered]
I have been using `usersConnection` to manage paginations with this schema in prisma1:
```graphql
"""A connection to a list of items."""
type UserConnection {
  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """A list of edges."""
  edges: [UserEdge]!
  aggregate: AggregateUser!
}

"""An edge in a connection."""
type UserEdge {
  """The item at the end of the edge."""
  node: User!

  """A cursor for use in pagination."""
  cursor: String!
}

type AggregateUser {
  count: Int!
}
```

now, it's time to migrate to prisma2. I have to keep this format as different frontend are using this schema, Any recommendations to recreate this pagination with prisma2? Should we parsed an re-create the object `users`? Is there a better approach? example:

```ts
    const users = await ctx.prisma.users.findMany({
      where: args.where,
      skip: args.skip,
      take: args.first,
    })


    const dataNode = data.map(singleData => {
      return {
        node: singleData
      }
    })
    return {
      edges: dataNode,
      aggregate: {
        count: 77 // should get the total
      }
    }


```

**Comment by ryands17** [comment]
Hey @alan345 👋 
If what you require is cursor based pagination, Prisma already supports [this](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/pagination#cursor-based-pagination) although I don't think it will match your existing GraphQL schema. If you really need to send it as per the above, you can follow the above that you performed and return the count via `prisma.users.count` method present.

Does this help?

**Comment by alan345** [comment, marked as answer]
thanks!
it this works for me:

```
  const users = await ctx.prisma.user.findMany({
      where: args.where,
      skip: args.skip,
      take: args.first,

    })
    return {
      edges: users.map((singleData: Source) => { return { node: singleData } }),
      aggregate: {
        count: await ctx.prisma.user.count({ where: args.where })
      }
    }

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
